### PR TITLE
sig-network, Make sig-network job mandatory and always run

### PR DESCRIPTION
**Pull request by oshoval**
This PR makes the `sig-network` job mandatory and always run.
It also removes sig-network tests from
`pull-kubevirt-e2e-k8s-1.19` job,
and from k8s-1.19 periodic job (note its name is `periodic-kubevirt-e2e-k8s-latest`).

It removes CNAO jobs, since `sig-network` runs all the tests that are needed now anyway,
see https://github.com/kubevirt/kubevirt/pull/5354 which moved the `VMIlifecycle` networking tests
to `sig-network`.

No test run duplication needed, and it will ease running
`sig-network` group isolated.

The job was tested periodically for a week, each 8 hours [1],
23 runs, all passed, beside two which are not due to the `sig-network`
tests problems [2] [3].
It's stable enough for this change.

Reduce timeout from 7h to 4h for `pull-kubevirt-e2e-k8s-1.19-sig-network`.

[1] https://prow.apps.ovirt.org/job-history/gs/kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.19-sig-network
[2] https://prow.apps.ovirt.org/view/gs/kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.19-sig-network/1373786699879944192
Failed due to https://github.com/kubevirt/kubevirt/issues/5285
[3] https://prow.apps.ovirt.org/view/gs/kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.19-sig-network/1374149087418388482#1:build-log.txt%3A49
Failed due to general quay connectivity error.

Signed-off-by: Or Shoval <oshoval@redhat.com>

### CODE REVIEW

**Comment by oshoval** [comment]
Idea to consider:
Since we are installing CNAO in sig-network lane, we can ditch CNAO lane, and move the tests it runs and sig-network didn't to the job (not to label them as sig-network because it means we will remove them from other jobs), just to run them in sig-network job.

**Comment by oshoval** [comment]
/cc @EdDev
please take a look

**Comment by oshoval** [comment]
Do we want to stop the periodic of `sig-network` ?

**Comment by oshoval** [comment]
It doesn't catch the filter good
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/1081/rehearsal-pull-kubevirt-e2e-k8s-1.19/1375001328866037760

**Comment by EdDev** [comment]
> Do we want to stop the periodic of sig-network ?

Not in this PR.

> It doesn't catch the filter good
> https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/1081/rehearsal-pull-kubevirt-e2e-k8s-1.19/1375001328866037760

Please elaborate. If this PR is not yet ready for review, please move it to draft.

General note: We have been asked by @fgimenez to wait with this until new HW is added to the CI cluster (mid next week).
Until then we can continue with the preparation (review, tests, etc).

**Comment by oshoval** [comment]
It has a bug, moving to draft and investigating

**Comment by oshoval** [comment]
Fixed the problem, when using ENV, need to double quote it, since '\\[' is used and its the one that should be used with double quote.

Rebased.

Ready for review, thanks.

**Review by EdDev — APPROVED** [review]

**Comment by EdDev** [comment]
/cc @fgimenez 

We are waiting for the new CI HW to be deployed in order to take this in.

**Comment by fgimenez** [comment]
> We are waiting for the new CI HW to be deployed in order to take this in.

I'll update here when the new machines are able to run jobs.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml
Comment (fgimenez) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:123]: Let's run this job in the new cluster
```suggestion
    optional: false
    cluster: prow-workloads
```

**Comment by oshoval** [comment]
Rebased
Addressed comments - run on new cluster
Thanks

**Comment by oshoval** [comment]
/rehearse

/hold

added dummy echo in order to rehearse

**Comment by oshoval** [comment]
Cluster should work already on not yet ?
Seems it doesn't schedule

https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/1081/rehearsal-pull-kubevirt-e2e-k8s-1.19-sig-network/1379352734217539585

**Comment by fgimenez** [comment]
We need more changes on the job:
* Disable docker proxy: set https://github.com/kubevirt/project-infra/blob/c0090737c14a42f64f61ed84cf6d8ce807501476/github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml#L133 to false
* Disable bazel cache: set https://github.com/kubevirt/project-infra/blob/c0090737c14a42f64f61ed84cf6d8ce807501476/github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml#L135 to false

Not sure about the untested bazel, to start with better to set it to false too https://github.com/kubevirt/project-infra/blob/c0090737c14a42f64f61ed84cf6d8ce807501476/github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml#L136

**Comment by oshoval** [comment]
Done thanks

**Comment by oshoval** [comment]
Great, it runs, 

It fails on 
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/1081/rehearsal-pull-kubevirt-e2e-k8s-1.19-sig-network/1379363180458807297

`09:27:52: e modprobe: FATAL: Module ip6_tables not found in directory /lib/modules/4.18.0-240.15.1.el8_3.x86_64`

We had it in the past, we might need to enable ipv6 on the new CI node please ?
https://linuxconfig.org/redhat-8-enable-disable-ipv6

and make sure the node has this module installed
since https://github.com/kubevirt/kubevirtci/pull/445 uses it

**Comment by oshoval** [comment]
Just touched the last commit so we can rehearse, for some reason it didnt reherase

**Comment by oshoval** [comment]
we have a problem 
https://github.com/kubevirt/kubevirt/issues/5113#issuecomment-814092722

**Comment by oshoval** [comment]
Now https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/1081/rehearsal-pull-kubevirt-e2e-k8s-1.19-sig-network/1379449369928929280 is passed

@EdDev 
The 2 tests that failed [1] one time are XFAIL flaky in the new cluster (sometime pass sometime XFAL)
Since there is a XFAIL still that can't be removed yet according what we spoke, i believe its not just due to the new cluster?

PR https://github.com/kubevirt/kubevirt/pull/5169 should consider to test them also on the new cluster i think ?

[1] https://github.com/kubevirt/kubevirt/issues/5113#issuecomment-814092722

**Comment by oshoval** [comment]
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/1081/rehearsal-pull-kubevirt-e2e-k8s-1.19-sig-network/1379449369928929280

The 2 tests fails once again, it might be something specific to the new cluster, even that they are XFAIL

**Comment by oshoval** [comment]
Rebased

**Comment by oshoval** [comment]
@fgimenez 
do we have this config on the new node ?
i think we might need it ?

https://github.com/kubevirt/project-infra/blob/master/github/ci/shared-deployments/files/docker-daemon-mirror.conf#L5

**Comment by fgimenez** [comment]
> do we have this config on the new node [1] ?

AFAIK this is the config for the docker-mirror service that runs in the cluster. Looking at one of the old SR-IOV nodes i see that `/etc/docker/daemon.conf` only contains `{}` and `/lib/systemd/system/docker.service` and `/etc/systemd/system/docker.service.d/custom.conf` don't contain any reference to `--fixed-cidr-v6` either.

**Comment by oshoval** [comment]
> > do we have this config on the new node [1] ?
> 
> AFAIK this is the config for the docker-mirror service that runs in the cluster. Looking at one of the old SR-IOV nodes i see that `/etc/docker/daemon.conf` only contains `{}` and `/lib/systemd/system/docker.service` and `/etc/systemd/system/docker.service.d/custom.conf` don't contain any reference to `--fixed-cidr-v6` either.

But its not sriov node,
I am trying to add preset-docker-mirror as usually we have either this one or the proxy one which gives those setting we need i think, wdyt?

**Comment by fgimenez** [comment]
Sure, the label for docker-mirror should work

**Comment by oshoval** [comment]
Thanks, running on a draft PR to test it without noising this one.

**Comment by oshoval** [comment]
Ran here
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/1115/rehearsal-pull-kubevirt-e2e-k8s-1.19-sig-network/1379782440788168704

they are failing, because as we discussed offline (correct me please if i am wrong), the mirror service is not there actually,
as a consequences we don't have those definitions we need on the new cluster
```
DOCKER_OPTS="${DOCKER_OPTS} --ipv6"
DOCKER_OPTS="${DOCKER_OPTS} --fixed-cidr-v6 2001:db8:1::/64"
```

There is a plan according @fgimenez to adapt the docker-mirror-proxy on the new machines.
I will open an issue and add a ref to the code.

Until then we have couple options:

1. Stay on the old machine with sig-network, we will ditch CNAO, and make sig-network the one that actually run so its one for one, until the new machine have proxy.
2. Move the sig-network to the new machine, stay with XFAIL, until the new machine have a proxy.
3. Implement some temporal solution for that, for example updating the docker config by the job (not sure it worths it?).

Please suggest which flow to take
@fgimenez @EdDev

**Comment by EdDev** [comment]
> There is a plan according @fgimenez to adapt the docker-mirror-proxy on the new machines.

What is the timeline for that?
But generally speaking, I am not sure why it is coupled to "mirror proxy", this setting is something we need to depend on (for now at least).

> Until then we have couple options (at least those, maybe more):
> 
>     Stay on the old machine with sig-network, we will ditch CNAO, and make sig-network the one that actually run so its one for one, until the new machine have proxy.
>     Move the sig-network to the new machine, stay with XFAIL, until the new machine have a proxy.
>     Implement some temporal solution for that, for example updating the docker config by the job (not sure it worths it?).

Based on the information provided, option 2 says we will not check it at all. I am ok with that for a limited timeline, so if this is going to be fixed in a reasonable timeline, we can go for it. But I think we need @AlonaKaplan approval for dropping the test.

**Comment by AlonaKaplan** [comment]
> > There is a plan according @fgimenez to adapt the docker-mirror-proxy on the new machines.
> 
> What is the timeline for that?
> But generally speaking, I am not sure why it is coupled to "mirror proxy", this setting is something we need to depend on (for now at least).
> 
> > Until then we have couple options (at least those, maybe more):
> > ```
> > - Stay on the old machine with sig-network, we will ditch CNAO, and make sig-network the one that actually run so its one for one, until the new machine have proxy.
> > - Move the sig-network to the new machine, stay with XFAIL, until the new machine have a proxy.
> > - Implement some temporal solution for that, for example updating the docker config by the job (not sure it worths it?).
> > ```
> 
> Based on the information provided, option 2 says we will not check it at all. I am ok with that for a limited timeline, so if this is going to be fixed in a reasonable timeline, we can go for it. But I think we need @AlonaKaplan approval for dropping the test.

Based on the information I read here, I would choose option 1.
Once the new cluster will be ready with the docker-mirror-proxy we can move to it.

**Comment by oshoval** [comment]
> Based on the information I read here, I would choose option 1.
> Once the new cluster will be ready with the docker-mirror-proxy we can move to it.

Thanks Alona, I also vote for that,
@fgimenez @EdDev wdyt please?

**Comment by fgimenez** [comment]
> I also vote for that, we ditch the CNAO job anyhow, so it won't add new jobs for the old cluster.

Ok, sounds good to me too, but the same PR should remove the CNAO presubmit lane and make the sig-network lane mandatory.

**Comment by oshoval** [comment]
Removed CNAO jobs.
Reverted moving sig-network to new cluster, will open other PR for that, which will be on hold until the new cluster docker mirror proxy is ready.

**Comment by oshoval** [comment]
/hold cancel
No reason to hold anymore (we are not moving to new cluster in this PR, because the ipv6 docker config is still missing there)

**Review by EdDev — COMMENTED** [review]
Thanks!

> - "export TARGET=k8s-1.19-sig-network && automation/test.sh"

Please consider formatting this like the others.

**Comment by EdDev** [comment]
@oshoval , @fgimenez , will this have any impact on the last branched branch?

**Comment by oshoval** [comment]
> @oshoval , @fgimenez , will this have any impact on the last branched branch?

Not that i know of, the periodic uses the main branch.
The presubmit jobs are not of releases

**Comment by fgimenez** [comment]
> will this have any impact on the last branched branch?

I don't think so, the presubmit lane will run on PRs only and has this setting:
```
skip_branches:
- release-\d+\.\d+
```

**Comment by fgimenez** [comment]
/lgtm
/approve

thanks!

**Comment by EdDev** [comment]
> Thanks!
> 
> > * "export TARGET=k8s-1.19-sig-network && automation/test.sh"
> 
> Please consider formatting this like the others.

Waiting for this answer.
/hold

**Comment by oshoval** [comment]
> > Thanks!
> > > * "export TARGET=k8s-1.19-sig-network && automation/test.sh"
> > 
> > 
> > Please consider formatting this like the others.
> 
> Waiting for this answer.
> /hold

will do that on a following PR if its ok

**Comment by oshoval** [comment]
> will do that on a following PR if its ok

https://github.com/kubevirt/project-infra/pull/1122

**Pull request merged**
